server/balancer.go

```diff
@@ -22,8 +23,30 @@ import (
 
 var storeCacheInterval = 30 * time.Second
 
+// minBalanceDiff returns the minimal diff to do balance. The formula is based
+// on experience to let the diff increase alone with the count slowly.
+func minBalanceDiff(count uint64) float64 {
+	return math.Pow(2, math.Log10(float64(count)))
+}
+
+// adjustBalanceSpeed returns a schedule limit and whether we should balance.
+// First, we calculate the diffCount (the number of schedules to balance the
+// source and the target). Then, we adjust the schedule limit to less than the
+// diffCount, so we will not over-scheduled. Finally, the min balance diff
+// provides a buffer to make the cluster stable, so we don't need to schedule
+// very frequently.
+func adjustBalanceSpeed(sourceCount uint64, sourceScore, targetScore float64) (uint64, bool) {
+	if targetScore >= sourceScore {
+		return 0, false
+	}
+	diffRatio := 1 - targetScore/sourceScore
+	diffCount := diffRatio * float64(sourceCount)
+	return uint64(diffCount) / 4, diffCount > minBalanceDiff(sourceCount)
```

Comment:
Well, if we want to balance two store at once, we can use `diffCount / 2` here, but consider that the store information may not up to date and the store feedback may delay, a smaller limit can help to reduce the effect of stale information. I choose `diffCount / 4` according to my experience, we may need to add a config to experiment it.